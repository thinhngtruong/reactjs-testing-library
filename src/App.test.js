import { render, screen, cleanup } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});

it("should equal to 0", () => {
  render(<App />);
  expect(screen.getByTestId("counter")).toHaveTextContent(0);
});

it("should be enabled", () => {
  render(<App />);
  expect(screen.getByTestId("button-up")).not.toHaveAttribute("disabled");
});

it("should be disabled", () => {
  render(<App />);
  expect(screen.getByTestId("button-down")).toBeDisabled();
});
