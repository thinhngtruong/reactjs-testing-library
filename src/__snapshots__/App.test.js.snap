// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      data-testid="counter"
    >
      0
    </h1>
    <button
      data-testid="button-up"
    >
       Up
    </button>
    <button
      data-testid="button-down"
      disabled=""
    >
      Down
    </button>
  </div>
</DocumentFragment>
`;
